src/render/md.jl

```diff
@@ -30,14 +31,184 @@ function validate(mdstyle::Dict{Symbol, ByteString})
     end
 end
 
-function save(file::String, mime::MIME"text/md", doc::Metadata, mdstyle::Dict{Symbol, ByteString} = DEFAULT_MDSTYLE;
-                                                                                mathjax = false, include_internal = true)
+"Permalink Entry used when collection needed info."
+type PLEntry
+    objname::ByteString
+    idname::ByteString
+    descr::ByteString
+    relsourcepath::ByteString
+end
+
+"Permalink Main helper type."
+type PLMain
+    genidxfile::ByteString
+    mdoutfile::ByteString      # reference to the current proccessed output file needed for relative links.
+    idxrelpath::ByteString     # current relative path: mdoutfile to genidxfile
+    headerstyle::ByteString
+    modnamestyle::ByteString
+    data::Dict{ByteString, Dict{Symbol, Array{PLEntry,1}}}
+end
+
+"""
+Return a relative filepath.
+
+Relative from `startpath` to `path`.
+This is a path computation:
+the filesystem is not accessed to confirm the existence or nature of `path` or `startpath`.
+
+**Example**
+
+```julia
+relpath_ = relpath("/home/user/Downloads", pwd())
+
+```
+"""
+function relpath(path::ByteString, startpath::ByteString)
+    pardir = ".."
+    if isabspath(path) && !contains(path, "/.")
+        path_arr = split(path, Base.path_separator_re)
+    else
+        path_arr = split(abspath(path), Base.path_separator_re)
+    end
+    if isabspath(startpath) && !contains(path, "/.")
+        start_arr = split(startpath, Base.path_separator_re)
+    else
+        start_arr = split(abspath(startpath), Base.path_separator_re)
+    end
+
+    count = 0
+    while count < min(length(path_arr), length(start_arr))
+        count += 1
+        if path_arr[count] != start_arr[count]
+            count -= 1
+            break
+        end
+    end
+    rel_path = "..$(Base.path_separator)"^(length(start_arr)-count) * join(path_arr[count+1:end],
+                                                                             Base.path_separator)
+    return rel_path
+end
```

Comment:
I assume this method works on Windows (don't have one to try it on)? Tests for it would be good to have.